File: Code/new_fitervari_backend/src/main/java/at/htl/boundary/WorkoutHistoryResource.java
package at.htl.boundary;

import at.htl.control.WorkoutHistoryRepository;
import at.htl.model.WorkoutHistory;

import javax.inject.Inject;
import javax.transaction.Transactional;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

@Path("/api/workoutHistory")
public class WorkoutHistoryResource {
    @Inject
    private WorkoutHistoryRepository repository;

    @GET
    @Path("/{id}")
    @Produces(MediaType.APPLICATION_JSON)
    public Response getById(@PathParam("id") Long id) {
        return Response.ok(repository.findById(id)).build();
    }

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Response getAll() {
        return Response.ok(repository.findAll().list()).build();
    }

    @POST
    @Transactional
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    public Response post(WorkoutHistory entity) {
        try{
            return Response.ok(repository.save(entity)).build();
        }catch (Exception e){
            return Response.serverError().build();
        }
    }

    @PUT
    @Transactional
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    public Response put(WorkoutHistory entity) {
        try{
            return Response.ok(repository.save(entity)).build();
        }catch (Exception e){
            return Response.serverError().build();
        }
    }

    @DELETE
    @Transactional
    @Path("/{id}")
    public Response delete(@PathParam("id") Long id) {
        try{
            repository.delete(repository.findById(id));
            return Response.ok().build();
        }catch (Exception e){
            return Response.serverError().build();
        }
    }
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/boundary/WorkoutHistoryResource.kt
package at.htl_leonding.boundary

import at.htl_leonding.service.WorkoutHistoryService
import javax.inject.Inject
import javax.ws.rs.*
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/api")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
class WorkoutHistoryResource {

    @Inject
    lateinit var service: WorkoutHistoryService

    @GET
    @Path("/workoutHistory/{id}")
    fun getExerciseById(@PathParam("id")id: Long): Response {
        return Response.ok(service.getById(id)).build()
    }
}